lib/web_ui/lib/src/engine/html/shader_mask.dart

```diff
@@ -0,0 +1,376 @@
+      break;
+    case ui.BlendMode.plus:
+      // Porter duff source + destination.
+      svgFilter = _compositeImageToSvg(imageUrl, 0, 1, 1, 0);
+      break;
+    case ui.BlendMode.modulate:
+      // Porter duff source * destination but preserves alpha.
+      svgFilter = _modulateImageToSvg(imageUrl);
+      break;
+    case ui.BlendMode.overlay:
+      // Since overlay is the same as hard-light by swapping layers,
+      // pass hard-light blend function.
+      svgFilter = _blendImageToSvg(imageUrl, 'hard-light', swapLayers: true);
+      break;
+    // Several of the filters below (although supported) do not render the
+    // same (close but not exact) as native flutter when used as blend mode
+    // for a background-image with a background color. They only look
+    // identical when feBlend is used within an svg filter definition.
+    //
+    // Saturation filter uses destination when source is transparent.
+    // cMax = math.max(r, math.max(b, g));
+    // cMin = math.min(r, math.min(b, g));
+    // delta = cMax - cMin;
+    // lightness = (cMax + cMin) / 2.0;
+    // saturation = delta / (1.0 - (2 * lightness - 1.0).abs());
+    case ui.BlendMode.saturation:
+    case ui.BlendMode.colorDodge:
+    case ui.BlendMode.colorBurn:
+    case ui.BlendMode.hue:
+    case ui.BlendMode.color:
+    case ui.BlendMode.luminosity:
+    case ui.BlendMode.multiply:
+    case ui.BlendMode.screen:
+    case ui.BlendMode.overlay:
+    case ui.BlendMode.darken:
+    case ui.BlendMode.lighten:
+    case ui.BlendMode.colorDodge:
+    case ui.BlendMode.colorBurn:
+    case ui.BlendMode.hardLight:
+    case ui.BlendMode.softLight:
+    case ui.BlendMode.difference:
+    case ui.BlendMode.exclusion:
+      svgFilter = _blendImageToSvg(imageUrl, _stringForBlendMode(blendMode));
+      break;
+    case ui.BlendMode.src:
+    case ui.BlendMode.dst:
+    case ui.BlendMode.dstATop:
+    case ui.BlendMode.dstIn:
+    case ui.BlendMode.dstOut:
+    case ui.BlendMode.dstOver:
+    case ui.BlendMode.clear:
+    case ui.BlendMode.srcOver:
+      assert(
+          false,
+          'Invalid svg filter request for blend-mode '
+          '$blendMode');
+      break;
+  }
+  return svgFilter;
+}
+
+int _maskFilterIdCounter = 0;
+
+// The color matrix for feColorMatrix element changes colors based on
+// the following:
+//
+// | R' |     | r1 r2 r3 r4 r5 |   | R |
+// | G' |     | g1 g2 g3 g4 g5 |   | G |
+// | B' |  =  | b1 b2 b3 b4 b5 | * | B |
+// | A' |     | a1 a2 a3 a4 a5 |   | A |
+// | 1  |     | 0  0  0  0  1  |   | 1 |
+//
+// R' = r1*R + r2*G + r3*B + r4*A + r5
+// G' = g1*R + g2*G + g3*B + g4*A + g5
+// B' = b1*R + b2*G + b3*B + b4*A + b5
+// A' = a1*R + a2*G + a3*B + a4*A + a5
+String _srcInImageToSvg(String imageUrl) {
+  _maskFilterIdCounter += 1;
+  return '<svg width="0" height="0" xmlns:xlink="http://www.w3.org/1999/xlink">'
+      '<filter id="_fmf$_maskFilterIdCounter" '
+      'filterUnits="objectBoundingBox">'
```

Comment:
`objectBoundingBox` is the default. We don't need to specify it. Will save a bit of CPU/memory/network.

(applies here and in other filters)

Reply:
Please test in Firefox. There are some cross browser bugs around filterUnits